### DISCUSSION TITLE
Advice on making a Swift framework.

### DISCUSSION THREAD

**Discussion by zamderax** [capnproto/capnproto#2138] [Q&A]
I am very excited about Cap'n Proto, A little dismayed that I can't find any Swift support, my next project will be effectively cross-platform Swift and I'm interested in all features like the RPC protocol as well as the zero-copy serialization.

A couple of questions:

1. Does anyone have any advice on how to build a Swift framework or SDK? 
    a. Is it enough to just wrap the C++ SDK in ergonomic Swift API functions and then work on the code generation? 
    b. Or should I write everything from scratch?
2. Is anyone interested in working on an open source implementation of Cap'n Proto? I'm happy to explore even a contract position to accomplish this.

**Comment by kentonv** [comment]
FWIW there was a past thread on Swift (when we still used Google Groups): https://groups.google.com/g/capnproto/c/29mX3BMhNbo?pli=1

But I'm not sure where that effort ended up.

I don't know much about Swift, but I know the Rust and Go implementations both translated/rewrote the entire runtime library into their respective languages. That said, on my project there's now desire to make the Rust implementation able to reuse the C++ RPC implementation (since it's very complicated and significantly more advanced than the Rust-native one), so there may be something to be said for layering on top of C++ to some extent.

**Reply by EliRibble** [reply]
I'm interested in Swift support as well. I'm currently in the process of deciding whether I think C++ interop or pure-Swift in the way to go. I'm looking at [danappelxx's start](https://github.com/Danappelxx/capnpc-swift) for inspiration/effort saving.
